### ISSUE TITLE
CalciteInputDatePicker with range sets end date when changing the start date if there's an initial value set

### ISSUE THREAD

**Issue by fnquinteros** [Esri/calcite-design-system#10113]
### Check existing issues

- [X] I have [checked for existing issues](https://github.com/Esri/calcite-design-system/issues) to avoid duplicates

### Actual Behavior

This only happens when there's an initial value set and only on the first input.

When it has an initial value set, setting the start date changes the end date and leaves the start date as it is.
If you refresh and start over, changing the end date first set the start date to null.

After the first value change, the input behaves es axpected.

### Expected Behavior

Setting the start or end date for the first time when there's an initial value changes them accordingly and do not change anything else.

### Reproduction Sample

https://codesandbox.io/p/devbox/happy-hooks-742ty6?file=%2Fsrc%2FApp.jsx

### Reproduction Steps

1. Change the start date to some day a few days in the future (3 or 4 days)
2. Observe the actual value set to the start date and check that the value you set was actually set on the end date.
3. refresh
4. set the end date
5. observe the start date. It will be empty.

### Reproduction Version

latest (2.11.1)

### Relevant Info

Tested on Chrome latest version, on a Macbook. Also ran on code sandbox, so it should happen regardless of OS. 

### Priority impact

impact - p1 - need for current milestone

### Impact

At the moment, this can be solved with a workaround by setting the initial value after rendering the component, instead of passing the initial value to the component right away.

We're planning on releasing a redesign of a feature that includes this date picker in October. Would be nice to have it fixed by then.

### Calcite package

- [ ] @esri/calcite-components
- [ ] @esri/calcite-components-angular
- [X] @esri/calcite-components-react
- [ ] @esri/calcite-design-tokens
- [ ] @esri/eslint-plugin-calcite-components

### Esri team

ArcGIS Urban

**Comment by DitwanP** [comment]
Moved to Nov milestone and increased priority at the request of Anvesh because an issue with react surfaced while verifying.

**Comment by anveshmekala** [comment]
While solving the issue ,we introduced a new bug with https://github.com/Esri/calcite-design-system/pull/8402 which affects `calcite-components-react` build. This is related to component lifecycle structure. Synced-up with @maxpatiiuk and this will be resolved once the Lit migration is complete. 

cc @DitwanP @jcfranco @maxpatiiuk

**Comment by anveshmekala** [comment]
This should be fixed in `3.0.0-next.51` version of `calcite-components-react`. 

https://codesandbox.io/p/devbox/input-date-picker-10113-kjynw6

cc @DitwanP , @fnquinteros

**Comment by DitwanP** [comment]
@anveshmekala I'm not able to verify with the link you provided, I'm getting this on codesandbox:
![image](https://github.com/user-attachments/assets/ae485d04-5038-41d4-a53f-da921ca9aa41)

**Comment by anveshmekala** [comment]
> @anveshmekala I'm not able to verify with the link you provided, I'm getting this on codesandbox: ![image](https://private-user-images.githubusercontent.com/60022782/396302515-ae485d04-5038-41d4-a53f-da921ca9aa41.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ0MTQxNTgsIm5iZiI6MTczNDQxMzg1OCwicGF0aCI6Ii82MDAyMjc4Mi8zOTYzMDI1MTUtYWU0ODVkMDQtNTAzOC00MWQ0LWE1M2YtZGE5MjFjYTlhYTQxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE3VDA1MzczOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyYzZlNmMxODIxY2Q3OWVjMzFjODcwNzY3NDIwZmViMGNjYWZlZDgwMDFlY2VjYzE2ZDliNDE4M2MwOGM5MjcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Okg5YcAFhb6xVTUHNiFT2vdQIzyI3DqS4QxKNc8SUhQ)

Apologies, sandbox is marked as private earlier. Should be accessible now.

**Comment by geospatialem** [comment]
Verified with https://codesandbox.io/p/devbox/input-date-picker-10113-kjynw6 in `@esri/calcite-components-react@3.0.0-next.51` 🎉
